src/Spatial/Euclidean/Line2D.cs

```diff
@@ -99,8 +110,29 @@ public Line2D LineTo(Point2D p, bool mustStartBetweenAndEnd)
                     dotProduct = l;
             }
 
-            Vector2D alongVector = dotProduct*this.Direction;
-            return new Line2D(this.StartPoint + alongVector, p);
+            Vector2D alongVector = dotProduct * this.Direction;
+            return this.StartPoint + alongVector;
+        }
+
+        /// <summary>
+        /// Compute the intersection between two lines
+        /// </summary>
+        /// <param name="other">The other line to compute the intersection with</param>
+        /// <returns>The point at the intersection of two lines, or null if the lines are parallelnu.</returns>
+        public Point2D? IntersectWith(Line2D other)
+        {
+            // http://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
+            Point2D p = this.StartPoint;
+            Point2D q = other.StartPoint;
+            Vector2D r = this.StartPoint.VectorTo(this.EndPoint);
+            Vector2D s = other.StartPoint.VectorTo(other.EndPoint);
+
+            double t = (q - p).CrossProduct(s) / (r.CrossProduct(s));
```

Comment:
`IsParallelTo(LIne2D other)` could perhaps be a separate method. Always tricky with tolerances for stuff like this.